packages/react/src/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,158 @@
+import * as RadixCheckbox from "@radix-ui/react-checkbox";
+import * as RadixLabel from "@radix-ui/react-label";
+import { type ComponentPropsWithRef, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { Text } from "../text";
+import * as styles from "./Checkbox.css";
+
+type CheckboxProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  {
+    checked?: "indeterminate" | boolean;
+    defaultChecked?: "indeterminate" | boolean;
+    disabled?: boolean;
+    helperText?: string;
+    label: string;
+    readonly?: boolean;
+  }
+>;
+
+export const Checkbox = forwardRef<HTMLDivElement, CheckboxProps>(
+  (
+    {
+      checked,
+      className,
+      defaultChecked,
+      disabled,
+      helperText,
+      id,
+      label,
+      readonly = false,
+      ...props
+    },
+    ref,
+  ) => {
+    const checkboxId =
+      id || `checkbox-${label.replace(/\s+/g, "-").toLowerCase()}`;
+
+    return (
+      <Flex
+        aria-disabled={disabled || readonly}
+        data-disabled={disabled || readonly}
```

Comment:
Using the Radix given `disabled` state, we can't make the keyboard interaction, if we want to make the component focusable through the keyboard, we can't use Radix inbuilt `disabled` . So here `aria-disabled` does nothing in fact.